libs/accounts/src/lib/account-card.tsx

```diff
@@ -35,19 +40,79 @@ const Button = ({
   onClick,
   label,
   icon,
+  ...props
 }: {
   onClick?: () => void;
   label: string;
   icon: VegaIconNames;
-}) => (
-  <TradingButton size="custom" className="p-[6px]" onClick={onClick}>
+} & ComponentProps<typeof TradingButton>) => (
+  <TradingButton size="custom" className="p-[6px]" onClick={onClick} {...props}>
     <div className="flex flex-col items-center p">
       <VegaIcon name={icon} size={16} className="mb-1" />
       <span className="text-xs">{label}</span>
     </div>
   </TradingButton>
 );
 
+const CombinedButton = ({
```

Comment:
Not sure why we need all this logic. Why not just render the dropdown and the button where needed? You may need to ensure the ref is passed down to <Button>

```
      {expandable && expanded ? (
        <div className="grid gap-1 grid-cols-4 p-3 pt-0">
          <TradingDropdown
            trigger={
              <TradingDropdownTrigger asChild>
                <Button label={t('Deposit')} icon={VegaIconNames.DEPOSIT} />
              </TradingDropdownTrigger>
            }
          >
            <TradingDropdownContent side="bottom" align="start">
              <TradingDropdownItem
                onClick={() => {
                  actions.onClickDeposit?.(asset.id);
                }}
              >
                <VegaIcon name={VegaIconNames.DEPOSIT} /> {t('Deposit')}
              </TradingDropdownItem>
              <TradingDropdownItem
                onClick={() => {
                  actions.onClickCrossChainDeposit?.(asset.id);
                }}
              >
                <VegaIcon name={VegaIconNames.DEPOSIT} />{' '}
                {t('Cross-chain deposit')}
              </TradingDropdownItem>
            </TradingDropdownContent>
          </TradingDropdown>
          <Button
            data-testid="account-action-swap"
            onClick={() => actions.onClickSwap?.(asset.id)}
            label={t('Swap')}
            icon={VegaIconNames.SWAP}
          />
          <Button
            data-testid="account-action-transfer"
            onClick={() => actions.onClickTransfer?.(asset.id)}
            label={t('Transfer')}
            icon={VegaIconNames.TRANSFER}
          />
          <Button
            data-testid="account-action-withdraw"
            onClick={() => actions.onClickWithdraw?.(asset.id)}
            label={t('Withdraw')}
            icon={VegaIconNames.WITHDRAW}
          />
        </div>
      ) : null}

```

Reply:
Just a leftover from the actual combined button, will clean this up shortly